wcfsetup/install/files/lib/acp/form/PackageUpdateServerAddForm.class.php

```diff
@@ -31,153 +41,167 @@ class PackageUpdateServerAddForm extends AbstractForm
+                            if (!empty($formField->getValidationErrors())) {
+                                return;
+                            }
+
+                            try {
+                                $url = new Uri($formField->getValue());
+
+                                if (!$url->getHost()) {
+                                    throw new UserInputException('serverURL', 'invalid');
+                                }
+                                if ($url->getHost() !== 'localhost') {
+                                    if ($url->getScheme() !== 'https') {
+                                        $formField->addValidationError(new FormFieldValidationError(
+                                            'invalidScheme',
+                                            'wcf.acp.updateServer.serverURL.error.invalidScheme',
+                                        ));
+                                    }
+                                    if ($url->getPort()) {
+                                        $formField->addValidationError(new FormFieldValidationError(
+                                            'nonStandardPort',
+                                            'wcf.acp.updateServer.serverURL.error.nonStandardPort',
+                                        ));
+                                    }
+                                }
+                                if ($url->getUserInfo()) {
+                                    $formField->addValidationError(new FormFieldValidationError(
+                                        'userinfo',
+                                        'wcf.acp.updateServer.serverURL.error.userinfo',
+                                    ));
+                                }
+                                if (\str_ends_with(\strtolower($url->getHost()), '.woltlab.com')) {
+                                    $formField->addValidationError(new FormFieldValidationError(
+                                        'woltlab',
+                                        'wcf.acp.updateServer.serverURL.error.woltlab',
+                                    ));
+                                }
+                            } catch (\InvalidArgumentException) {
+                                $formField->addValidationError(new FormFieldValidationError(
+                                    'invalid',
+                                    'wcf.acp.updateServer.serverURL.error.invalid',
+                                ));
+                            }
+
+                            if (
+                                ($duplicate = $this->getDuplicateServer((string)$url))
+                                && ($this->formAction === IFormDocument::FORM_MODE_CREATE
+                                || ($this->formObject instanceof PackageUpdateServer
+                                && $this->formObject->getObjectID() !== $duplicate->getObjectID()))
```

Comment:
The indentation is super wonky here. One additional indentation level per opening parentheses, please. I believe the following is correct:
```suggestion
                                ($duplicate = $this->getDuplicateServer((string)$url))
                                && (
                                    $this->formAction === IFormDocument::FORM_MODE_CREATE
                                    || (
                                        $this->formObject instanceof PackageUpdateServer
                                        && $this->formObject->getObjectID() !== $duplicate->getObjectID()
                                    )
                                )
```